fuzzing/valuegenerationtracer/valuegeneration_tracer.go

```diff
@@ -0,0 +1,357 @@
+			outputValues = append(outputValues, outputValue)
+		}
+	}
+
+	return outputValues
+}
+
+// captureExitedCallFrame is a helper method used when a call frame is exited, to record information about it.
+func (v *ValueGenerationTracer) captureExitedCallFrame(output []byte, err error) {
+	// If this was an initial deployment, now that we're exiting, we'll want to record the finally deployed bytecodes.
+	if v.currentCallFrame.ToRuntimeBytecode == nil {
+		// As long as this isn't a failed contract creation, we should be able to fetch "to" byte code on exit.
+		if !v.currentCallFrame.IsContractCreation() || err == nil {
+			v.currentCallFrame.ToRuntimeBytecode = v.evm.StateDB.GetCode(v.currentCallFrame.ToAddress)
+		}
+	}
+	if v.currentCallFrame.CodeRuntimeBytecode == nil {
+		// Optimization: If the "to" and "code" addresses match, we can simply set our "code" already fetched "to"
+		// runtime bytecode.
+		if v.currentCallFrame.CodeAddress == v.currentCallFrame.ToAddress {
+			v.currentCallFrame.CodeRuntimeBytecode = v.currentCallFrame.ToRuntimeBytecode
+		} else {
+			v.currentCallFrame.CodeRuntimeBytecode = v.evm.StateDB.GetCode(v.currentCallFrame.CodeAddress)
+		}
+	}
+
+	// Resolve our contract definitions on the call frame data, if they have not been.
+	v.resolveCallFrameContractDefinitions(v.currentCallFrame)
+
+	// Set our information for this call frame
+	v.currentCallFrame.ReturnData = slices.Clone(output)
+	v.currentCallFrame.ReturnError = err
+
+	if v.currentCallFrame.ReturnError == nil {
+		// Grab return data of the call frame
+		v.trace.transactionOutputValues = append(v.trace.transactionOutputValues, v.getCallFrameReturnValue())
+	}
+
+	// We're exiting the current frame, so set our current call frame to the parent
+	v.currentCallFrame = v.currentCallFrame.ParentCallFrame
+}
+
+func (v *ValueGenerationTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, scope *vm.ScopeContext, rData []byte, depth int, err error) {
+	// TODO: look for RET opcode (for now try getting them from currentCallFrame.ReturnData)
+	// Execute all "on next capture state" events and clear them.
+	for _, eventHandler := range v.onNextCaptureState {
+		eventHandler()
+	}
+	v.onNextCaptureState = nil
+
+	// If a log operation occurred, add a deferred operation to capture it.
+	if op == vm.LOG0 || op == vm.LOG1 || op == vm.LOG2 || op == vm.LOG3 || op == vm.LOG4 {
+		v.onNextCaptureState = append(v.onNextCaptureState, func() {
+			logs := v.evm.StateDB.(*state.StateDB).Logs()
+			if len(logs) > 0 {
+				v.currentCallFrame.Operations = append(v.currentCallFrame.Operations, logs[len(logs)-1])
+			}
+		})
+	}
+}
+
+func (v *ValueGenerationTracer) CaptureFault(pc uint64, op vm.OpCode, gas, cost uint64, scope *vm.ScopeContext, depth int, err error) {
+	//TODO implement me
+	//panic("implement me")
+}
+
+func AddTransactionOutputValuesToValueSet(results *types.MessageResults, valueSet *valuegeneration.ValueSet) {
```

Comment:
make this function the responsibility of the `ValueSet` and not tracer. The set can accept a list of `[]any` and use a switch statement, similar to here, to determine which subset to add the value to.